packages/server-renderer/src/renderToString.ts

```diff
@@ -142,12 +144,21 @@ function renderComponentSubTree(
       setCurrentRenderingInstance(null)
     } else if (comp.render) {
       renderVNode(push, renderComponentRoot(instance), instance)
+    } else if (comp.template && isString(comp.template)) {
```

Comment:
I think we can move this whole branch to the top with

``` js
if (!comp.ssrRender && !comp.render && isString(comp.template)) {
  comp.ssrRender = ...
}

if (comp.ssrRender) { ... }
```

So that we don't have to duplicate the `ssrRender` logic twice.